src/Microsoft.ML.StandardLearners/Standard/SdcaStatic.cs

```diff
@@ -0,0 +1,156 @@
+            };
+
+            var rec = new TrainerEstimatorReconciler.Regression(
+                (env, labelName, featuresName, weightsName) =>
+                {
+                    var trainer = new SdcaRegressionTrainer(env, args, featuresName, labelName, weightsName);
+                    if (onFit != null)
+                        return trainer.WithOnFitDelegate(trans => onFit(trans.Model));
+                    return trainer;
+                }, label, features, weights);
+
+            return rec.Score;
+        }
+
+        public static (Scalar<float> score, Scalar<float> probability, Scalar<bool> predictedLabel)
+            PredictSdcaBinaryClassification(this Scalar<bool> label, Vector<float> features, Scalar<float> weights = null,
+                float? l2Const = null,
+                float? l1Threshold = null,
+                float convergenceTolerance = 0.1f,
+                int? maxIterations = null,
+                bool shuffle = true,
+                float biasLearningRate = 0,
+                Action<LinearBinaryPredictor, ParameterMixingCalibratedPredictor> onFit = null)
+        {
+            var args = new LinearClassificationTrainer.Arguments()
+            {
+                L2Const = l2Const,
+                L1Threshold = l1Threshold,
+                ConvergenceTolerance = convergenceTolerance,
+                MaxIterations = maxIterations,
+                Shuffle = shuffle,
+                BiasLearningRate = biasLearningRate
+            };
+
+            var rec = new TrainerEstimatorReconciler.BinaryClassifier(
+                (env, labelName, featuresName, weightsName) =>
+                {
+                    var trainer = new LinearClassificationTrainer(env, args, featuresName, labelName, weightsName);
+                    if (onFit != null)
+                    {
+                        return trainer.WithOnFitDelegate(trans =>
+                        {
+                            // Under the default log-loss we assume a calibrated predictor.
+                            var model = trans.Model;
+                            var cali = (ParameterMixingCalibratedPredictor)model;
+                            var pred = (LinearBinaryPredictor)cali.SubPredictor;
+                            onFit(pred, cali);
+                        });
+                    }
+                    return trainer;
+                }, label, features, weights);
+
+            return rec.Output;
+        }
+
+        private sealed class TrivialFactory : ISupportSdcaClassificationLossFactory
+        {
+            private readonly ISupportSdcaClassificationLoss _loss;
+
+            public TrivialFactory(ISupportSdcaClassificationLoss loss)
+            {
+                _loss = loss;
+            }
+
+            public ISupportSdcaClassificationLoss CreateComponent(IHostEnvironment env)
+            {
+                // REVIEW: We are ignoring env?
```

Comment:
>REVIEW: We are ignoring env? [](start = 19, length = 28)

well, in this case the loss has been given to us, so no need to use env to manufacture anything, right?